app/models/spree/order.rb

```diff
@@ -434,6 +434,7 @@ def empty!
       payments.clear
       shipments.destroy_all
       restart_checkout_flow if state == "payment"
+      restart_checkout_flow if state == "confirmation"
```

Comment:
can probably combine this with line 436

`restart_checkout_flow if ["payment", "confirmation"].include?(state)`

Reply:
Thanks @georgethoppil, I've combined it following your suggestion. Wondering if other states should not be included here as well. I've left it like this for now though.